cloudinit/config/cc_puppet.py

```diff
@@ -237,21 +239,28 @@ def handle(name: str, cfg: Config, cloud: Cloud, args: list) -> None:
         )
 
         if install_type == "packages":
+            to_install: List[Union[str, List[str]]]
             if package_name is None:  # conf has no package_nam
                 for puppet_name in PUPPET_PACKAGE_NAMES:
-                    try:
-                        cloud.distro.install_packages((puppet_name, version))
+                    with suppress(InstallerError):
```

Comment:
Smart approach w/ the suppress handling as we loop over the install_packages calls.